### DISCUSSION TITLE
missing `getTrialsOfType` from jsPsych.data

### DISCUSSION THREAD

**Discussion by teonbrooks** [jspsych/jsPsych#388] [Q&A]
Hey, I was working with the example from the jspsych tutorial and the received an error in the console mentioned that `jsPsych.data` doesn't have the method `getTrialsOfType`. I inspected it and couldn't find a comparable method within the object. Is there a suggested substitute?

**Comment by teonbrooks** [comment]
also, an additional questions, is jQuery a requirement? I couldn't find it being called anywhere but it seems to be still suggested in the tutorial.

**Comment by jodeleeuw** [comment]
The `master` branch of GitHub is the development version. The docs at docs.jspsych.org refer to the last release, which is still v5.0.3. That's causing this mismatch between the tutorial and the website. The `/docs` folder in the repo is partially updated to reflect v6.0 changes, but there are still lots of updates to the docs that will occur before the official release.

For `getTrialsofType` the new way is to use the more generic and powerful `filter` method:

```
var all_single_stim_trials = jsPsych.data.get().filter({trial_type: 'single-stim'});
var the_array_of_data = all_single_stim_trials.values(); // need to call .values() now because the filter() method returns a DataCollection object.
```

jQuery is no longer required for v6.0.

**Comment by teonbrooks** [comment]
awesome, that's great :) thanks!
